pandas/tests/arrays/string_/test_string.py

```diff
@@ -101,7 +101,7 @@ def test_add(dtype, request):
             "unsupported operand type(s) for +: 'ArrowStringArray' and "
             "'ArrowStringArray'"
         )
-        mark = pytest.mark.xfail(raises=TypeError, reason=reason)
+        mark = pytest.mark.xfail(raises=NotImplementedError, reason=reason)
```

Comment:
BTW (not necessarily for this PR), but `+` for strings is actually implemented in Arrow, but under the "join" name (like str.join), so this could be emulated using `pc.binary_join_element_wise(left, right, "")` (the last empty string is to not use a separator)